backend/src/services/enhanced_guide_service.py

```
@@ -970,16 +980,22 @@ async def _generate_with_real_searches(self, context: Dict, progress_callback: O
             
             # Enhance restaurants with photos and booking URLs
-            if restaurants:
-                restaurants = await self.places_enhancer.enhance_restaurants_batch(
-                    restaurants, context["destination"]
-                )
+            if restaurants and isinstance(restaurants, list):
+                try:
+                    restaurants = await self.places_enhancer.enhance_restaurants_batch(
+                        restaurants, context["destination"]
+                    )
+                except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)